testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -161,6 +161,31 @@
   raise ScriptError, "Checkbox #{identifier} not unchecked." if has_checked_field?(identifier)
 end
 
+When(/^I (check|uncheck) "([^"]*)" by label$/) do |action, identifier|
+  begin
+    checkbox = find(:xpath, "//input[@type='checkbox' and @id = //label[normalize-space(.) = '#{identifier}']/@for]")
```

Comment:
```suggestion
    checkbox = find(:xpath, ""//label[text()='#{label}']/preceding-sibling::input[@type='checkbox']"")
```

![image](https://github.com/user-attachments/assets/7acfe7a0-94c8-4ff9-9b6a-74c6a9091d1d)

Reply:
okey